CoEpi/repo/realm/RealmCENKeyRepo.swift

```diff
@@ -1,36 +1,39 @@
-import Foundation
 import CryptoKit
+import Foundation
 import RealmSwift
 import Security
 
-struct CENKey : Codable {
-    var timestamp: Int64 = Int64(Date().timeIntervalSince1970)
-    var cenKey: String?
-    
+class RealmCENKeyRepo: CENKeyRepo, RealmRepo {
+    var realm: Realm
+
     //static var cenKey: String = ""
-    static var cenKeyTimestamp: Int64 = 0
-    
-    static func generateAndStoreCENKey() -> CENKey {
+    var cenKeyTimestamp: Int64 = 0
+
+    init(realmProvider: RealmProvider) {
+        realm = realmProvider.realm
+    }
+
+    func generateAndStoreCENKey() -> CENKey {
         //Retrieve last cenKey and cenKeyTimestamp from CENKey
         let latestCENKey = getLatestCENKey()
         let curTimestamp = Int64(Date().timeIntervalSince1970)
         if ( ( cenKeyTimestamp == 0 ) || ( roundedTimestamp(ts: curTimestamp) > roundedTimestamp(ts: cenKeyTimestamp) ) ) {
             //generate a new AES Key and store it in local storage
-            
+
             //generate base64string representation of key
             let cenKeyString = computeSymmetricKey()
             let cenKeyTimestamp = curTimestamp
-            
+
             //Create CENKey and insert/save to Realm
-            let newCENKey = CENKey(timestamp: cenKeyTimestamp, cenKey: cenKeyString)
-            newCENKey.insert()
+            let newCENKey = CENKey(cenKey: cenKeyString, timestamp: cenKeyTimestamp)
+            let _ = insert(key: newCENKey)
             return newCENKey
         } else {
             return latestCENKey!
```

Comment:
We have to handle these optionals properly, this seems unsafe.

Reply:
Not really sure what the default behavior should be...

Reply:
Yeah let's keep the code "as is" for now.